### PR TITLE
fix #882. Test cases added for HttpSyncDataConfiguration.

### LINKED ISSUE #882: Test Spring-boot-starter-sync-data-http

### PR DESCRIPTION
**Pull request by strawberry-crisis**
Fixes issue #882

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml [hunk 1]
```diff
@@ -38,5 +38,22 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+            <version>3.11.0</version>
```
Comment (tydhot) [inline soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml:49]: okhttp's version has been declared in root pom.xml

**Review by tydhot — DISMISSED** [review]

File: soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml (hunk 1 above)
Comment (strawberry-crisis) [reply soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml:49]: image
The declared version of okhttp leads to building failure.

Should I refactor these tests with alternatives?

Comment (dengliming) [reply soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml:49]: Yes. you can try `WireMockRule`. There is a example [here](https://github.com/dromara/soul/blob/master/soul-client/soul-client-common/src/test/java/org/dromara/soul/client/common/utils/OkHttpToolsTest.java).

Comment (strawberry-crisis) [reply soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml:49]: Thx a lot, I'll fix it.

File: soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml [hunk 2]
```diff
@@ -38,5 +38,20 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>okhttp</artifactId>
+        </dependency>
```
Comment (dengliming) [inline soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/pom.xml:49]: we can also remove this

File: soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/src/test/java/org/dromara/soul/spring/boot/starter/sync/data/http/HttpClientPluginConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+             BufferedReader bufferedReader = new BufferedReader(reader)
+        ) {
+            StringBuilder builder = new StringBuilder();
+            bufferedReader.lines().forEach(builder::append);
+            return builder.toString();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
```
Comment (dengliming) [inline soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/src/test/java/org/dromara/soul/spring/boot/starter/sync/data/http/HttpClientPluginConfigurationTest.java:125]: Maybe you can simplify here like this.

```
 private static String mockConfigsFetchResponseJson() throws Exception {
        return new String(Files.readAllBytes(
                Paths.get(Objects.requireNonNull(HttpClientPluginConfigurationTest.class.getClassLoader()
                        .getResource("mock_configs_fetch_response.json")).toURI())));
    }
```

Comment (strawberry-crisis) [reply soul-spring-boot-starter/soul-spring-boot-starter-sync-data-center/soul-spring-boot-starter-sync-data-http/src/test/java/org/dromara/soul/spring/boot/starter/sync/data/http/HttpClientPluginConfigurationTest.java:125]: gladly

**Review by dengliming — APPROVED** [review]

**Pull request merged**